### ISSUE TITLE
C# ParseTreePatternMatcher.Compile fails

### ISSUE THREAD

**Issue by appel1** [antlr/antlr4#2097]
ParserInterpreter creates an instance of ParserATNSimulator with null for the decisionToDFA parameter. ParserATNSimulator.AdaptivePredict throws a NullReferenceException if it is null. 

```
grammar Test;

options {
    language=CSharp;
}

text: TEXT* EOF;

TEXT: [a-z]+;

WS: [ \t\r\n]+ -> skip;
```

```csharp
var lexer = new TestLexer(new AntlrInputStream("text"));
var parser = new TestParser(new CommonTokenStream(lexer));
var context = parser.text();

var pattern = parser.CompileParseTreePattern("<TEXT>", TestParser.RULE_text);
```

```
Antlr4.Runtime.Tree.Pattern.ParseTreePatternMatcher+CannotInvokeStartRule: Object reference not set to an instance of an object. ---> System.NullReferenceException: Object reference not set to an instance of an object.
   at Antlr4.Runtime.Atn.ParserATNSimulator.AdaptivePredict(ITokenStream input, Int32 decision, ParserRuleContext outerContext)
   at Antlr4.Runtime.ParserInterpreter.VisitState(ATNState p)
   at Antlr4.Runtime.ParserInterpreter.Parse(Int32 startRuleIndex)
   at Antlr4.Runtime.Tree.Pattern.ParseTreePatternMatcher.Compile(String pattern, Int32 patternRuleIndex)
   --- End of inner exception stack trace ---
   at Antlr4.Runtime.Tree.Pattern.ParseTreePatternMatcher.Compile(String pattern, Int32 patternRuleIndex)
   at Antlr4.Runtime.Parser.CompileParseTreePattern(String pattern, Int32 patternRuleIndex, Lexer lexer)
   at Antlr4.Runtime.Parser.CompileParseTreePattern(String pattern, Int32 patternRuleIndex)
   at ConsoleApp3.Program.Main(String[] args) in ConsoleApp3\ConsoleApp3\Program.cs:line 19
```

**Comment by ericvergnaud** [comment]
Is this with the official tool/runtime or the tunnel vision labs one?

Envoyé de mon iPhone

**Comment by appel1** [comment]
This is with the Antlr4.Runtime.Standard v4.7.1 NuGet package. 

Relevant parts of the code:
https://github.com/antlr/antlr4/blob/master/runtime/CSharp/runtime/CSharp/Antlr4.Runtime/ParserInterpreter.cs#L65
https://github.com/antlr/antlr4/blob/master/runtime/CSharp/runtime/CSharp/Antlr4.Runtime/Atn/ParserATNSimulator.cs#L317

**Comment by sharwell** [comment]
@ericvergnaud The answer is the same as last time. Your `AntlrInputStream` type isn't in my releases.

**Comment by appel1** [comment]
@xied75 No, we skipped using tree parser so I never got around to trying to fix it myself.

**Comment by sharwell** [comment]
@parrt This can be closed (it was fixed by #2382)
